builtin/providers/cloudstack/provider.go

```diff
@@ -72,10 +88,42 @@ func Provider() terraform.ResourceProvider {
 }
 
 func providerConfigure(d *schema.ResourceData) (interface{}, error) {
+	apiURL := d.Get("api_url").(string)
+	apiKey := d.Get("api_key").(string)
+	secretKey := d.Get("secret_key").(string)
+
+	if configFile := d.Get("config").(string); configFile != "" {
```

Comment:
I think it's cleaner to check for this like so:

```go
if configFile, ok := d.GetOk("config"); ok {
  config, err := ini.Load(configFile.(string))
<snip>
```
This would give an error if you did set the `config` param, but assigned it an empty string (the `ini.Load()` func would return an error in that case). I think that would be more clear for users.